File: android/app/src/main/java/com/zahraag/pawsitivehabits/data/models/UserSettings.kt
package com.zahraag.pawsitivehabits.data.models

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName
import kotlinx.serialization.Serializable
import kotlinx.serialization.Transient
import org.bson.types.ObjectId

@Serializable
@Entity("userSettings_table")
data class UserSettings(
    @PrimaryKey @SerializedName("_id")
    val id: String = org.bson.types.ObjectId().toHexString(),
    var userId: String,
    var language: String = "en",
    var weightUnit: String = "kg",
    var notificationsEnabled: Boolean = true,
    var biometricLockEnabled: Boolean = false,

    @kotlinx.serialization.Transient
    var isSynced: Boolean = false
)

File: api/src/models/userSettings.js
const mongoose = require('mongoose');

const UserSettingsSchema = new mongoose.Schema({
    _id: { type: String, required: true },
    userId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true,
        unique: true,
        index: true
    },
    language: { type: String, default: 'en' },
    weightUnit: { type: String, default: 'kg' },
    notificationsEnabled: { type: Boolean, default: true },
    biometricLockEnabled: { type: Boolean, default: false }
}, { timestamps: true });

module.exports = mongoose.model('UserSettings', UserSettingsSchema);